internal/db/schema_test.go

```
@@ -123,6 +128,62 @@ func TestInitUpgradesLegacyScheduleSchema(t *testing.T) {
+		// user 7 is gone; foreign keys were off when it was deleted, so its rows survived.
+		`INSERT INTO device (id, name, os, hash, user_id) VALUES (1, 'Kept', 'test', 'kept', 1), (2, 'Orphan', 'test', 'orphan', 7)`,
+		`INSERT INTO list (id, name, user_id) VALUES (1, 'Kept', 1), (16, 'Orphan', 7)`,
+		`INSERT INTO schedule (id, name, start_time, end_time, days, active, user_id) VALUES (1, 'Kept', '08:00', '17:00', 31, 1, 1), (2, 'Orphan', '08:00', '17:00', 31, 1, 7)`,
+		`INSERT INTO list_rule (list_id, rule_id) VALUES (1, 1), (16, 1)`,
+		`INSERT INTO schedule_rule (schedule_id, rule_id) VALUES (1, 1), (2, 1)`,
+		`INSERT INTO list_schedule (list_id, schedule_id) VALUES (16, 2)`,
+		`INSERT INTO device_schedule (device_id, schedule_id) VALUES (2, 2)`,
+	)
+	for _, query := range queries {
+		if _, err := legacy.Exec(query); err != nil {
+			legacy.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `legacy.Close` is not checked

(errcheck)